cheatsheets/Server_Side_Request_Forgery_Prevention_Cheat_Sheet.md

```diff
@@ -20,6 +22,27 @@ SSRF is an attack vector that abuses an application to interact with the interna
 * SSRF is not limited to the HTTP protocol, despite the fact that in general the first request leverages it, yet the second request is performed by the application itself, and thus it could be using different protocols (*e.g.* FTP, SMB, SMTP, etc.) and schemes (*e.g.* `file://`, `phar://`, `gopher://`, `data://`, `dict://`, etc.). The protocol/scheme usage is highly dependent on the application's requirements.
 * If the application is vulnerable to [XML eXternal Entity (XXE) injection](https://portswigger.net/web-security/xxe) then it can by exploited to perform a [SSRF attack](https://portswigger.net/web-security/xxe#exploiting-xxe-to-perform-ssrf-attacks), take a look at the [XXE cheat sheet](XML_External_Entity_Prevention_Cheat_Sheet.md) to learn how to prevent the exposure to XXE.
 
+## Contents
+
+- [Cases](#cases)
+  * [Case 1 - Application can send request only to identified and trusted applications](#case-1---application-can-send-request-only-to-identified-and-trusted-applications)
+    + [Example](#example)
+    + [Available protections](#available-protections)
+      - [Application layer](#application-layer)
+        * [String](#string)
+        * [IP address](#ip-address)
+        * [Domain name](#domain-name)
+        * [URL](#url)
+      - [Network layer](#network-layer)
+  * [Case 2 - Application can send requests to ANY external IP address or domain name](#case-2---application-can-send-requests-to-any-external-ip-address-or-domain-name)
+    + [Challenges in blocking URLs at application layer](#challenges-in-blocking-urls-at-application-layer)
+    + [Available protections](#available-protections-1)
+      - [Application layer](#application-layer-1)
+      - [Network layer](#network-layer-1)
+- [IMDSv2 in AWS](#imdsv2-in-aws)
+- [References](#references)
+- [Tools and code used for schemas](#tools-and-code-used-for-schemas)
+
 # Cases
```

Comment:
Should this (and subsequent headings) all be h2, if the title is h1? Somethign to standardise.